src/scvmm/azext_scvmm/_params.py

```diff
@@ -0,0 +1,265 @@
+            c.argument(
+                'dynamic_memory_enabled',
+                arg_type=get_three_state_flag(),
+                help='If dynamic memory should be enabled.',
+            )
+            c.argument(
+                'dynamic_memory_min',
+                type=int,
+                options_list=['--dynamic-memory-min'],
+                help="DynamicMemoryMin in MBs for the vm.",
+            )
+            c.argument(
+                'dynamic_memory_max',
+                type=int,
+                options_list=['--dynamic-memory-max'],
+                help="DynamicMemoryMax in MBs for the vm.",
+            )
+            c.argument(
+                'availability_sets',
+                options_list=['--availability-sets', '-a'],
+                nargs='+',
+                help="List of the name or the ID of the availability sets for the vm.",
+            )
+
+    with self.argument_context('scvmm vm create') as c:
+        c.argument(
+            'vm_template',
+            options_list=['--vm-template', '-t'],
+            help="Name or ID of the vm template for deploying the vm.",
+        )
+        c.argument(
+            'cloud',
+            options_list=['--cloud', '-c'],
+            help="Name or ID of the cloud for deploying the vm.",
+        )
+        c.argument(
+            'admin_password',
+            options_list=['--admin-password'],
+            help="Admin password for the vm.",
+        )
+        c.argument(
+            'nics',
+            options_list=['--nic'],
+            action=VmNicAddAction,
+            nargs='+',
+            help="Network overrides for the vm."
+            "Usage: --nic name=<> network=<> ipv4-address-type=<> "
+            "ipv6-address-type=<> mac-address-type=<> mac-address=<>.",
+        )
+        c.argument(
+            'disks',
+            options_list=['--disk'],
+            action=VmDiskAddAction,
+            nargs='+',
+            help="Disk overrides for the vm."
+            "Usage: --disk name=<> disk-size=<> template-disk-id=<> bus-type=<> "
+            "bus=<> lun=<> vhd-type=<> qos-name=<> qos-id=<>.",
+        )
+
+    self.argument_context('scvmm vm update')
+
+    with self.argument_context('scvmm vm stop') as c:
+        c.argument(
+            'skip_shutdown',
+            action='store_true',
+            help="Skips shutdown and power-off immediately.",
+        )
+
+    with self.argument_context('scvmm vm nic') as c:
+        c.argument('nic_name', options_list=['--name', '-n'], help="Name of the NIC.")
+        c.argument('nic_id', options_list=['--nic-id'], help="UUID of the NIC.")
+        c.argument(
+            'vm_name', options_list=['--vm-name'], help="Name of the virtual machine."
+        )
+        c.argument(
+            'network',
+            options_list=['--network'],
+            help="Name or Id of the virtual network.",
+        )
+        c.argument(
+            'disconnect',
+            action='store_true',
```

Comment:
Why not use 'get_three_state_flag'?

Reply:
Have changed it to `get_three_state_flag`.